sync-gateway-config.json

```diff
@@ -97,10 +96,12 @@ function(doc, oldDoc){
     access(doc.owner, "task-list." + doc._id + ".users");
   } else if (getType() == "task") {
     /* Write Access */
+    var owner = doc._deleted ? oldDoc.taskList.owner : doc.taskList.owner;
```

Comment:
I feel like write permission first still makes sense, in particular around the error messages being returned.  For a user without write access, it's inappropriate to be returning field validation errors.

However, my point here also applies to the validation that's happening at https://github.com/couchbaselabs/mobile-training-todo/blob/master/sync-gateway-config.json#L109.  Attempting to access doc.taskList.id is going to throw an error if taskList isn't present, even before you reach the validateNotEmpty function.